### ISSUE TITLE
chatglm3-6b sft.sh过程中的acc指标和infer.sh后的acc差距很大，请问可能是什么原因？

### ISSUE THREAD

**Issue by kaer1990** [modelscope/swift#177]
### 场景
相关性识别
### 样本示例
{"instruction": "query：苹果手机\n title：iphone 6s，99新 \n请判断上述query和title是否相关？", "input": "", "output": "是"}
### sft.sh  参数
nproc_per_node=1
base_path="xxx"
train_data="xxx"
val_data="xxx"

PYTHONPATH=$base_path \
CUDA_VISIBLE_DEVICES=1 \
torchrun \
    --nproc_per_node=$nproc_per_node \
    --master_port 29500 \
    ${base_path}/examples/pytorch/llm/llm_sft.py \
    --model_id_or_path ZhipuAI/chatglm3-6b \
    --model_revision master \
    --sft_type lora \
    --tuner_backend swift \
    --template_type chatglm3 \
    --dtype fp16 \
    --output_dir output \
    --custom_train_dataset_path $train_data \
    --custom_val_dataset_path $val_data \
    --ddp_backend nccl \
    --train_dataset_sample -1 \
    --num_train_epochs 1 \
    --max_length 2048 \
    --check_dataset_strategy warning \
    --lora_rank 8 \
    --lora_alpha 32 \
    --lora_dropout_p 0.05 \
    --lora_target_modules DEFAULT \
    --gradient_checkpointing true \
    --batch_size 1 \
    --weight_decay 0.01 \
    --learning_rate 1e-4 \
    --gradient_accumulation_steps $(expr 16 / $nproc_per_node) \
    --max_grad_norm 0.5 \
    --warmup_ratio 0.03 \
    --eval_steps 100 \
    --save_steps 200 \
    --save_total_limit -1 \
    --logging_steps 10 \
    --push_to_hub false \
    --hub_model_id chatglm3-6b-lora \
    --hub_private_repo true \
    --hub_token 'your-sdk-token' \
    --only_save_model true \

### infer.sh 参数
base_path="xxx"

PYTHONPATH=$base_path \
CUDA_VISIBLE_DEVICES=0 \
python ${base_path}/examples/pytorch/llm/llm_infer.py \
    --ckpt_dir "xxx" \
    --load_args_from_ckpt_dir true \
    --eval_human false \
    --max_length 2048 \
    --max_new_tokens 100 \
    --temperature 0.1 \
    --top_k 5 \
    --top_p 0.7 \
    --repetition_penalty 1.05 \
    --do_sample false \
    --merge_lora_and_save false \
    --val_dataset_sample -1 \

### 问题
**sft 的logging.jsonl 文件中 acc 和 eval_acc 都有0.95左右， 但是infer.sh 预测的结果，计算acc只有0.88， 怎么回事？**

sft 的logging.jsonl 文件部分内容：
{"loss": 0.10637206, "acc": 0.9500001, "learning_rate": 9.07e-06, "epoch": 0.81, "global_step": 1110}
{"loss": 0.090618, "acc": 0.9604167, "learning_rate": 8.4e-06, "epoch": 0.82, "global_step": 1120}
{"loss": 0.09619076, "acc": 0.9697917, "learning_rate": 7.75e-06, "epoch": 0.83, "global_step": 1130}
{"loss": 0.12981621, "acc": 0.95625, "learning_rate": 7.13e-06, "epoch": 0.83, "global_step": 1140}
{"loss": 0.09293562, "acc": 0.95833321, "learning_rate": 6.53e-06, "epoch": 0.84, "global_step": 1150}
{"loss": 0.10500737, "acc": 0.9625, "learning_rate": 5.96e-06, "epoch": 0.85, "global_step": 1160}
{"loss": 0.11163101, "acc": 0.9604166, "learning_rate": 5.41e-06, "epoch": 0.86, "global_step": 1170}
{"loss": 0.08441702, "acc": 0.9687499, "learning_rate": 4.89e-06, "epoch": 0.86, "global_step": 1180}
{"loss": 0.09331145, "acc": 0.9635417, "learning_rate": 4.39e-06, "epoch": 0.87, "global_step": 1190}
{"loss": 0.12667795, "acc": 0.9562499, "learning_rate": 3.91e-06, "epoch": 0.88, "global_step": 1200}
{"eval_loss": 0.1006394, "eval_acc": 0.95512341, "eval_runtime": 127.4426, "eval_samples_per_second": 12.869, "eval_steps_per_second": 12.869, "epoch": 0.88, "global_step": 1200}

**Comment by Jintao-Huang** [comment]
sft时候的acc是包含special_token的, 并且不是step by step的acc. 😊
